src/GraphQL/Validation/Errors/ArgumentsOfCorrectTypeError.cs

```diff
@@ -14,16 +12,16 @@ public class ArgumentsOfCorrectTypeError : ValidationError
         /// <summary>
         /// Initializes a new instance with the specified properties.
         /// </summary>
-        public ArgumentsOfCorrectTypeError(ValidationContext context, Argument node, IEnumerable<string> verboseErrors)
-            : base(context.Document.OriginalQuery, NUMBER, BadValueMessage(node.Name, AstPrinter.Print(node.Value), verboseErrors), node)
+        public ArgumentsOfCorrectTypeError(ValidationContext context, Argument node, string verboseErrors)
+            : base(context.Document.OriginalQuery, NUMBER, BadValueMessage(node.Name, verboseErrors), node)
         {
         }
 
-        internal static string BadValueMessage(string argName, string value, IEnumerable<string> verboseErrors)
+        internal static string BadValueMessage(string argName, string verboseErrors)
         {
-            var message = verboseErrors != null ? $"\n{string.Join("\n", verboseErrors)}" : "";
-
-            return $"Argument \"{argName}\" has invalid value {value}.{message}";
+            return string.IsNullOrEmpty(verboseErrors)
+                ? $"Argument '{argName}' has invalid value."
+                : $"Argument '{argName}' has invalid value. {verboseErrors}";
```

Comment:
Simplify?  Either way is fine.
```suggestion
            return $"Argument '{argName}' has invalid value.{(string.IsNullOrEmpty(verboseErrors) ? null : " " + verboseErrors)}";
```

Reply:
This option is more difficult to read and there are more allocations in it, although they are not on the hot path.